### DISCUSSION TITLE
[Discussion] Why not MappingProxyType?

### DISCUSSION THREAD

**Discussion by lifelonglearner127** [Marco-Sulla/python-frozendict#51] [Q&A]
Thank you for writing this wonderful python package.
I don't think this is the right place to ask this question. However curious about this module. What are the differences from this package from **types.MappingProxyType**?
We can make the dictionary as read-only by using **MappingProxyType**?
Best Regards,
Levan Begashvili

**Comment by lurch** [comment]
I've never looked at MappingProxyType before, but https://stackoverflow.com/questions/41795116/difference-between-mappingproxytype-and-pep-416-frozendict might have some relevant info?

**Comment by lifelonglearner127** [comment]
Thanks for your quick reply.
so frozendict focus only on dict wherease MappingProxyType works as an interface, I think.
By the way, frozendict add immutability on the first-level just like MappingProxyType. So is it a known issue?

**Comment by Marco-Sulla** [comment]
I do not use `MappingProxyType` because is *really* slow. The old implementation used it. Since nothing stops you to change the old `_dict` attribute the contained the `MappingProxyType`, I didn't see any benefit from using it. 

Py `frozendict` is not really immutable, since, well, it's written in Python. With this new version is only a bit more complicated, since you can change it only using `object.__setattr__()`. `MappingProxyType` is really immutable, but slow.

If you want a real immutable frozendict, you want the C Extension. If you have Linux you can install one of the Linux packages: https://github.com/Marco-Sulla/python-frozendict/releases If you don't have Linux, you can clone the repo and launch `python3 setup.py c bdist_wheel` and install the resulting wheel.

I would create a VM with CentOS 5 and create a `manylinux` distro, so I can publish it to Pypi. Furthermore, I have to remove the `c` parameter from `setup.py` and make it the default (if someone wants the pure Python only, a `py` parameters would be used instead)
